xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java

```diff
@@ -144,21 +146,25 @@ public int hashCode() {
   }
 
   static final class CidrRange {
-    private final String addressPrefix;
+    private final InetAddress addressPrefix;
     private final int prefixLen;
 
     @VisibleForTesting
-    CidrRange(String addressPrefix, int prefixLen) {
-      this.addressPrefix = addressPrefix;
+    CidrRange(String addressPrefix, int prefixLen) throws InvalidProtocolBufferException {
+      try {
+        this.addressPrefix = InetAddress.getByName(addressPrefix);
+      } catch (UnknownHostException e) {
+        throw new InvalidProtocolBufferException(e.getMessage());
```

Comment:
I don't think `InvalidProtocolBufferException` is appropriate type for this error. It could be misleading.

Reply:
The description of [`InvalidProtocolBufferException`](https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/InvalidProtocolBufferException) says that "Thrown when a protocol message being parsed is invalid in some way, e.g. it contains a malformed varint or a negative byte length."

Now the `address_prefix` part of `CidrRange` is [described](https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/core/v3/address.proto#L156) as "IPv4 or IPv6 address, e.g. ``192.0.0.0`` or ``2001:db8::``." . So if this string contains anything other than an IPv4 or IPv6 address then `InetAddress.getByName()` throws `UnknownHostException` which should be transformed to `InvalidProtocolBufferException` and eventually the [code here](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/ServerXdsClient.java#L172) will send a NACK to the xDS control plane.

Reply:
Once `ServerXdsClient` is merged with `ClientXdsClient`, we might return `StructOrError` instead of throwing. C and Go are doing similarly for parsing and nacking. For now we can keep as-is.